src/modules/displaymanager/main.py

```diff
@@ -439,22 +442,30 @@ def run():
 
             if default_desktop_environment is not None:
                 os.system(
-                    "sed -i -e \"s/^.*user-session=.*/user-session={!s}/\" "
-                    "{!s}/etc/lightdm/lightdm.conf".format(
+                    "sed -i -e \"s/^.*user-session=.*/user-session={!s}/\" " "{!s}".format(
                         default_desktop_environment.desktop_file,
-                        root_mount_point
-                        )
-                    )
-
-            if default_desktop_environment.desktop_file == "deepin":
-                os.system(
-                    "sed -i -e \"s/^.greeter-session=.*"
-                    "/greeter-session=lightdm-deepin-greeter/\" "
-                    "{!s}/etc/lightdm/lightdm.conf".format(
-                        root_mount_point
+                        lightdm_conf_path
                         )
                     )
 
+            # configure lightdm-greeter
+            greeter_path = os.path.join(
+                root_mount_point, "usr/share/xgreeters"
+                )
+            if (os.path.exists(greeter_path)):
+                greeter = os.listdir(greeter_path)[0].split('.')[0]
```

Comment:
Right. Normally we only install one greeter. So we can add a check to see if it is a actually a **.desktop** file or not.